#! /usr/bin/env python


PACKAGE='longitudinal_controller'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        Name          	                Type        Level   Description			             Default Min	Max
gen.add("update_param_",       bool_t,   0,   "update_param",       False)
gen.add("distance_kp", 	double_t,	0,  "distance_kp", 		0.75,  0,  5.0)
gen.add("distance_ki", 	double_t,	0,  "distance_ki", 		0.0,	0,	5.0)
gen.add("distance_kd", 	double_t,	0,  "distance_kd", 		0.0,	0,	5.0)
gen.add("distance_i_max", 	double_t,	0,  "distance_i_max", 		0.5,	0,	10.0)
gen.add("distance_out_max", 	double_t,	0,  "distance_out_max", 5.0,	0,	10.0)
gen.add("speed_max",	int_t,		0,	"speed_max(km/h)",		40,		0,	50)
gen.add("speed_kp", 	double_t,	0,  "speed_kp", 		1.0,  0,  5.0)
gen.add("speed_ki", 	double_t,	0,  "speed_ki", 		0.0,	0,	5.0)
gen.add("speed_kd", 	double_t,	0,  "speed_kd", 		0.0,	0,	5.0)
gen.add("speed_i_max", 	double_t,	0,  "speed_i_max", 		0.5,	0,	10.0)
gen.add("speed_out_max", 	double_t,	0,  "speed_out_max", 	5.0,	0,	10.0)

exit(gen.generate(PACKAGE, "longitudinal_controller", "lon_controller"))
